### PR TITLE
Follow the documentation's recommendation

### PR DESCRIPTION
**Pull request by AvdN**
The doc states "Make sure to pass options explicitly.", but the example didn't. 
Since constructing a string for the shell with appropriate quotes is non-trivial, the first example should be how to use `execProcess`  including `args`

### CODE REVIEW

**Comment by c-blake** [comment]
Why not the simpler example of `execProcess("nim", ["c", "-r", "mytestfile.nim"])`?  This call uses `poUsePath` so there is no need to use an absolute path to `env` to ensure PATH search.  As conceived this example call suggests PATH search is not done which is misleading.

Also, if you try to compile *either* of those two examples they fail because `workingDir` is between `command` and `args` in the parameter list (probably not the best parameter order. IMO - cmd, args, wd would be better, though with keyword param calls it obv. doesn't matter).

Finally, if you do amend either the `env` form or simple form to, e.g., `execProcess("nim", "", ["c", "-r", "mytestfile.nim"])` they *still* fail because `poEvalCommand` is used by default.  In default-on `poEvalCommand` mode, `args` is not used at all - the call passes the first string to `/bin/sh -c`.

You can make examples of this pre-tokenized `args`-passing style work by altering the declared default options to *not* have `poEvalCommand` as in, `execProcess("nim", args=["c", "-r", "mytestfile.nim"], options={poStdErrToStdOut, poUsePath})`.  However, that's a pretty complex example.

TL;DR - the old example was better - it at least compiled and didn't mislead.

**Comment by AvdN** [comment]
@c-blake  I was not familiar with `poUsePath`, thanks for pointing that out. I just tried to figure out how to not have to do the complex string compositing for `ExecShellCmd`. 

I found the old example misleading, as it is not following the advice in the documentation to explicitly pass `options`, and above all useless because it would require compositing, while taking care of parameters with spaces, etc. Unfortunately, while copying the example from my test environment (which had a different command), I missed the `s` in `options`.

I am not sure why you indicate that both examples don't compile, the second one **is** the original example with a different variable name, and it does compile, as-is, for me using nim 0.19.4.

**Comment by c-blake** [comment]
There is no `ExecShellCmd` in the entire history of the Nim github.

Running any but the last of the repaired with `workingDir` versions that I mentioned raises an exception conditional from ``assert args.len == 0``.

Also, you misspelled the keyword parameter in your PR - you said "option" instead of "options".  So, your version should not cut & paste compile. :-)

I didn't try 0.19.4, but with a `devel` version of Nim I get `first type mismatch at position: 2` if I don't put an empty string in for `workingDir` which seems correct compiler rejection to me.  The two I meant in "either of these two" would be an `option`->`options` corrected version of yours and my first "simpler example" one.

If you really want the example to use an already tokenized `args seq` (which I agree is more robust general programming practice, especially if the string is built from any run-time inputs) then I suggest:
```Nim
import osproc
echo execProcess("nim", args=["c", "-r", "mytestfile.nim"], options={poUsePath})
```
which at least "works" end-to-end without compile or automatic run-time errors and without an unnecessary extra "env".  (I don't think you need the `import`/`echo` in the doc comment, but they are needed for a proper test.)

{ Well, as briefly alluded to, really I would suggest a new overload allowing `execProcess("nim", ["c", ..])` that maybe also has `options={poUsePath,poStdErrToStdOut}` as a default, but that may not be on the table/is more work. It's only a one..two-liner to make your own, though.}

**Comment by Araq** [comment]
Damn. :-)
